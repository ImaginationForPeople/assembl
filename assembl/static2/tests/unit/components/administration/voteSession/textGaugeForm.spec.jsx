import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { List, Map } from 'immutable';

import DumbTextGaugeForm from '../../../../../js/app/components/administration/voteSession/textGaugeForm';
import '../../../../helpers/setupTranslations';

describe('DumbTextGaugeForm component', () => {
  const handleGaugeChoiceLabelChange = jest.fn();

  it('should render a form to set up a textual gauge', () => {
    const choices = List.of(
      Map({
        id: 'choice-1',
        title: 'contextually-based'
      }),
      Map({
        id: 'choice-2',
        title: 'Savings Account Knoll'
      })
    );
    const props = {
      choices: choices,
      handleGaugeChoiceLabelChange: handleGaugeChoiceLabelChange
    };
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<DumbTextGaugeForm {...props} />);
    const result = shallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});