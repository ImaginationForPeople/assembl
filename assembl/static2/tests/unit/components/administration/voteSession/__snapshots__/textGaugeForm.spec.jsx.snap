// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DumbTextGaugeForm component should render a form to set up a textual gauge 1`] = `
<div>
  Immutable.List [
    <div>
      <FormControlWithLabel
        helperUrl=""
        label="Value title 1"
        labelAlwaysVisible={false}
        onChange={[Function]}
        required={true}
        type="text"
        validationErrors={null}
        value="contextually-based"
      />
    </div>,
    <div>
      <FormControlWithLabel
        helperUrl=""
        label="Value title 2"
        labelAlwaysVisible={false}
        onChange={[Function]}
        required={true}
        type="text"
        validationErrors={null}
        value="Savings Account Knoll"
      />
    </div>,
  ]
</div>
`;
